baking/src/tezos_baking/tezos_setup_wizard.py

```diff
@@ -405,21 +417,11 @@ def check_blockchain_data(self):
             return False
         return True
 
-    # Check the provider url and collect the most recent snapshot
-    # that is suited for the chosen history mode and network
-    def get_snapshot_metadata(self, name, json_url):
-        def hashes_comply(s1, s2):
-            return s1.startswith(s2) or s2.startswith(s1)
-
-        try:
-            snapshot_array = None
-            with urllib.request.urlopen(json_url) as url:
-                snapshot_array = json.load(url)["data"]
-            snapshot_array.sort(reverse=True, key=lambda x: x["block_height"])
-
-            node_version_hash = get_node_version_hash()
+    def extract_snapshot_metadata(self, snapshot_array):
```

Comment:
This definitely could use a comment.

The name suggests that this extracts metadata from a snapshot, but without reading its implementation one cannot tell that this does also the filtering and picking of the best snapshot available.

Also, the strategy used to pick the snapshot is not written down anywhere either, but it should be.